### ISSUE TITLE
ionic upload doesn't work

### ISSUE THREAD

**Issue by e-oz** [ionic-team/ionic-cli#2694]
#### Short description of the problem:
Command "ionic upload" doesn't upload anything

#### What behavior are you expecting?
Uploaded app in Ionic View

**Steps to reproduce:**
1. create new app
2. run ionic upload

**Which @ionic/app-scripts version are you using?**
ionic info

cli packages: (/Users/OZ/.config/yarn/global/node_modules)

    @ionic/cli-utils  : 1.9.2
    ionic (Ionic CLI) : 3.9.2

global packages:

    Cordova CLI : 6.5.0 

local packages:

    @ionic/app-scripts : 2.1.4
    Cordova Platforms  : android 6.2.3 ios 4.3.1
    Ionic Framework    : ionic-angular 3.6.0

System:

    ios-sim : 5.1.0 
    Node    : v8.0.0
    npm     : 5.4.0 
    OS      : macOS High Sierra
    Xcode   : Xcode 8.3.2 Build version 8E2002

**Other information:** 
ionic upload
[INFO] Running app-scripts build: 
       
[16:15:12]  build dev started ... 
[16:15:12]  clean started ... 
[16:15:12]  clean finished in 2 ms 
[16:15:12]  copy started ... 
[16:15:12]  transpile started ... 
[16:15:17]  transpile finished in 4.45 s 
[16:15:17]  preprocess started ... 
[16:15:17]  deeplinks started ... 
[16:15:17]  deeplinks finished in 153 ms 
[16:15:17]  preprocess finished in 154 ms 
[16:15:17]  webpack started ... 
[16:15:17]  copy finished in 4.85 s 
[16:15:28]  webpack finished in 10.76 s 
[16:15:28]  sass started ... 
[16:15:29]  sass finished in 1.24 s 
[16:15:29]  postprocess started ... 
[16:15:29]  postprocess finished in 9 ms 
[16:15:29]  lint started ... 
[16:15:29]  build dev finished in 16.67 s 
> ionic cordova prepare
> cordova prepare
✔ Running command - done!

[16:15:33]  lint finished in 3.76 s 

(and that's all)

**Comment by e-oz** [comment]
Just reproduced on the absolutely fresh-generated project and got message:

```
 (node:3609) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error [ERR_IPC_CHANNEL_CLOSED]: channel closed
(node:3609) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Commands were executed: 
ionic start uplTest
cd uplTest/
ionic upload

nothing else was touched.

**Comment by e-oz** [comment]
Could you please write any commentary about this issue? Is it bug or Ionic View will not be usable anymore?

**Comment by janunezc** [comment]
I have the same issue.

Rolled back to ionic version 3.5.0 using `npm install -g ionic@3.5`  and it works fine. So somewhere between 3.5 and 3.9 it got broken.

I have to say this is one of the most important features in ionic which make the difference when compared with its competitors. It should not stay broken.

**Comment by dwieeb** [comment]
Does this happen with `ionic upload --nobuild`?

**Comment by e-oz** [comment]
@dwieeb actually you can try it pretty easy - it comes out of the box with newly generated project, nothing needs to be changed to reproduce :) And answer is yes, it doesn't work even with "ionic upload  --nobuild". All usual things as updating npm don't help.

**Comment by e-oz** [comment]
It's strictly related to this issue:
https://github.com/ionic-team/ionic-cli/issues/2375
And currently can be fixed by upgrading Node to 8.4.0

**Comment by dwieeb** [comment]
@e-oz Oh, weird. Okay, thanks for following up. Other people have been having these issues with upload
